Fails:

> error: self-comparison always evaluates to true [-Werror=tautological-compare]

Could not reproduce this locally, so that might be something Arch's gcc has a patch for already.

Currently using `gcc-9 amd64 9.1.0-2ubuntu2~16.04`.

```
[139/262] Building C object src/nvim/CMakeFiles/nvim.dir/window.c.o
FAILED: /usr/lib/ccache/gcc-9 -DINCLUDE_GENERATED_DECLARATIONS -DMIN_LOG_LEVEL=3 -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -DUSE_GCOV -D_GNU_SOURCE -Iconfig -I../src -Isrc/nvim/auto -Iinclude -I/home/travis/nvim-deps/usr/include/luajit-2.0 -isystem /home/travis/nvim-deps/usr/include --coverage -g   -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fdiagnostics-color=always -Wno-array-bounds -Werror  -Wno-conversion -MD -MT src/nvim/CMakeFiles/nvim.dir/window.c.o -MF src/nvim/CMakeFiles/nvim.dir/window.c.o.d -o src/nvim/CMakeFiles/nvim.dir/window.c.o   -c ../src/nvim/window.c
../src/nvim/window.c: In function ‘do_window’:
../src/nvim/window.c:183:33: error: self-comparison always evaluates to true [-Werror=tautological-compare]
  183 |     FOR_ALL_WINDOWS_IN_TAB(wp2, curtab) {
      |                                 ^~
../src/nvim/window.c: In function ‘win_valid’:
../src/nvim/window.c:1504:30: error: self-comparison always evaluates to true [-Werror=tautological-compare]
 1504 |   FOR_ALL_WINDOWS_IN_TAB(wp, curtab) {
      |                              ^~
../src/nvim/window.c: In function ‘win_count’:
../src/nvim/window.c:1536:30: Kerror: self-comparison always evaluates to true [-Werror=tautological-compare]
 1536 |   FOR_ALL_WINDOWS_IN_TAB(wp, curtab) {
      |                              ^~
../src/nvim/window.c: In function ‘one_window’:
../src/nvim/window.c:2244:30: error: self-comparison always evaluates to true [-Werror=tautological-compare]
 2244 |   FOR_ALL_WINDOWS_IN_TAB(wp, curtab) {
      |                              ^~
../src/nvim/window.c: In function ‘buf_jump_open_win’:
../src/nvim/window.c:4411:32: error: self-comparison always evaluates to true [-Werror=tautological-compare]
 4411 |     FOR_ALL_WINDOWS_IN_TAB(wp, curtab) {
      |                                ^~
../src/nvim/window.c: In function ‘win_size_save’:
../src/nvim/window.c:4749:30: error: self-comparison always evaluates to true [-Werror=tautological-compare]
 4749 |   FOR_ALL_WINDOWS_IN_TAB(wp, curtab) {
      |                              ^~
../src/nvim/window.c: In function ‘win_size_restore’:
../src/nvim/window.c:4768:34: error: self-comparison always evaluates to true [-Werror=tautological-compare]
 4768 |       FOR_ALL_WINDOWS_IN_TAB(wp, curtab) {
      |                                  ^~
../src/nvim/window.c: In function ‘win_setminheight’:
../src/nvim/window.c:5223:32: error: self-comparison always evaluates to true [-Werror=tautological-compare]
 5223 |     FOR_ALL_WINDOWS_IN_TAB(wp, curtab) {
      |                                ^~
../src/nvim/window.c: In function ‘only_one_window’:
../src/nvim/window.c:5995:30: error: self-comparison always evaluates to true [-Werror=tautological-compare]
 5995 |   FOR_ALL_WINDOWS_IN_TAB(wp, curtab) {
      |                              ^~
../src/nvim/window.c: In function ‘win_id2win’:
../src/nvim/window.c:6691:30: error: self-comparison always evaluates to true [-Werror=tautological-compare]
 6691 |   FOR_ALL_WINDOWS_IN_TAB(wp, curtab) {
      |                              ^~
cc1: all warnings being treated as errors
[139/262] Building C object src/nvim/C...o/msgpack_lua_c_bindings.generated.c.o
ninja: build stopped: subcommand failed.
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -55,6 +55,34 @@ anchors:
     # adds the target branch to the cache key.
     FOR_TRAVIS_CACHE=$TRAVIS_BRANCH
 
+addons:
+  apt:
+    packages: &common-apt-packages
+      - apport
+      - autoconf
+      - automake
+      - build-essential
+      - clang-4.0
+      - cmake
+      - cscope
+      - gcc-multilib
+      - gdb
+      - gperf
+      - language-pack-tr
+      - libc6-dev-i386
+      - libtool-bin
+      - locales
+      - ninja-build
+      - pkg-config
+      - unzip
+      - valgrind
+      - xclip
+  homebrew:
+    update: true
+    packages:
+      - ccache
+      - ninja
+
 jobs:
   include:
     - stage: baseline
@@ -106,14 +134,22 @@ jobs:
       env:
         - *common-job-env
 
-    - name: gcc-coverage
+    - name: gcc-coverage (gcc 9)
       os: linux
-      compiler: gcc
+      compiler: gcc-9
       env:
         - GCOV=gcov
         - CMAKE_FLAGS="$CMAKE_FLAGS -DUSE_GCOV=ON"
         - GCOV_ERROR_FILE="/tmp/libgcov-errors.log"
         - *common-job-env
+      addons:
+        apt:
+          sources:
+            - sourceline: 'ppa:ubuntu-toolchain-r/test'
+          packages:
+            - *common-apt-packages
+            - gcc-9
+
     - name: clang-tsan
       os: linux
       compiler: clang
@@ -129,34 +165,6 @@ install:        ci/install.sh
 before_script:  ci/before_script.sh
 script:         ci/script.sh
 before_cache:   ci/before_cache.sh
-
-addons:
-  apt:
-    packages:
-      - apport
-      - autoconf
-      - automake
-      - build-essential
-      - clang-4.0
-      - cmake
-      - cscope
-      - gcc-multilib
-      - gdb
-      - gperf
-      - language-pack-tr
-      - libc6-dev-i386
-      - libtool-bin
-      - locales
-      - ninja-build
-      - pkg-config
-      - unzip
-      - valgrind
-      - xclip
-  homebrew:
-    update: true
-    packages:
-      - ccache
-      - ninja
 
 branches:
   only:

--- a/.travis.yml
+++ b/.travis.yml
@@ -46,8 +46,8 @@ env:
     - CCACHE_COMPRESS=1
     - CCACHE_SLOPPINESS=time_macros,file_macro
     - CCACHE_BASEDIR="$TRAVIS_BUILD_DIR"
-    # Default since 3.3, but Travis (Xenial) has 3.2.4; workaround (#10533).
-    - CCACHE_HASHDIR=1
+    # Default since 3.3, but Travis (Xenial) has 3.2.4; required with newer gcc/clang.
+    - CCACHE_CPP2=1
 
 anchors:
   envs: &common-job-env

